browser.js

```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+module.exports = function () {
+  throw new Error(
```

Comment:
I've read https://github.com/websockets/ws/issues/1344, but I am personally not sure why it needs to throw when it can fallback. It looks like we are waiting for @3rd-Eden 's input?

Maybe it's already discussed at other places, but should this [fallback to browser's `WebSocket`](https://github.com/heineiuo/isomorphic-ws/blob/master/browser.js), while announcing that `WebSocket.Server` doesn't work on browser?

I think that would make many projects' dependency graph a bit cleaner and more straightforward when they need to implement a platform agnostic web socket client.

Reply:
@shogowada this has been discussed ad nauseam, we don't want to maintain a browser shim because there are so many differences/edge cases that makes it a pain.

Refs: https://github.com/websockets/ws/commit/e54d45fbab3b19c2940e9057ce1e7b8f105873e0

Reply:
Thanks for the additional context.

In that case, instead of throwing (that appears to break some), would a warning work?

Something like this:

```javascript
module.exports = function () {
  // Almost a copy & paste from the README + link to the ref you gave me for future people like me🙂
  console.warn('ws module does not work in the browser. Browser clients must use the native WebSocket object. To make the same code work seamlessly on Node.js and the browser, you can use one of the many wrappers available on npm, like isomorphic-ws. For more context, see https://github.com/websockets/ws/commit/e54d45fbab3b19c2940e9057ce1e7b8f105873e0.')
}
```

Reply:
I don't think it helps, it still breaks existing code. Throwing an error is the right thing to do imho.